src/simulator.py

```diff
@@ -963,6 +980,16 @@ def simulation_aligned_metagenome(min_l, max_l, median_l, sd_l, out_reads, out_e
 
 def simulation_aligned_transcriptome(model_ir, out_reads, out_error, kmer_bias, basecaller, read_type, num_simulate,
                                      polya, fastq, per=False, uracil=False):
+
+    if basecaller == "albacore":
+        polya_len_dist_scale = 2.409858743694814
+    else:
+        polya_len_dist_scale = 4.168299657168961
+    
+    # inner function to return a polyA tail length    
+    def get_polya_len():
+        return int(scipy.stats.expon.rvs(loc=2.0, scale=polya_len_dist_scale))
+                                         
```

Comment:
@kmnip, first of all, thanks for implementing this. PolyA modeling is one of the desired but missing features in NanoSim. 

Are we planning to model the polya len distribution scale for different basecallers and library prep protocols in the future? How much difference are there between various methods? Have you done any analysis on this (to come up with these numbers)  and do you have any insight/thoughts on this?

Reply:
No, this isn't new at all. I just moved the code into a function so that the `basecaller` variable isn't checked in every iteration of the `while` loop.

https://github.com/bcgsc/NanoSim/blob/ff052d5634b1ffdfcfa170de6a6320792a1e421f/src/simulator.py#L1096-L1103

It looks like Chen and Theodora implemented it.
https://github.com/bcgsc/NanoSim/blame/ff052d5634b1ffdfcfa170de6a6320792a1e421f/src/simulator.py#L1098

Reply:
I don't think it should be tied to the basecallers. The polyA length distribution should be trained just like head/tail lengths. I would prefer that over what is hard-coded here.

Reply:
I totally agree with that. Perhaps something to work on in the future releases. But until then, what is confusing to me is whether those scale values should be different based on pore chemistry, cDNA vs directRNA sequencing, and etc? It looks like that it is currently tied to the basecaller type.